eth/gaspricemonitor.go

```diff
@@ -32,18 +32,25 @@ type GasPriceMonitor struct {
 	gasPriceMu sync.RWMutex
 	// gasPrice is the current gas price to be returned to users
 	gasPrice *big.Int
+	// minGasPrice is the minimum gas price below which polled values will be discarded
+	minGasPrice *big.Int
 
 	// update is a channel used to send notifications to a listener
 	// when the gas price is updated
 	update chan struct{}
 }
 
 // NewGasPriceMonitor returns a GasPriceMonitor
-func NewGasPriceMonitor(gpo GasPriceOracle, pollingInterval time.Duration) *GasPriceMonitor {
+func NewGasPriceMonitor(gpo GasPriceOracle, pollingInterval time.Duration, minGasPrice *big.Int) *GasPriceMonitor {
+	minGasP := big.NewInt(0)
```

Comment:
Don't think we need this as the minGasPrice param value is not from an untrusted source (unlike in other cases where we should be more defensive since an untrusted source could give us unexpected values) and the caller is responsible for passing in a sane value.

Reply:
I disagree, I don't think it's correct to assume that a caller will pass in sane values, ever.  It's not because we pass in a sane value that a third party potentially using this package will. Passing in a `nil` value will lead to unexpected behaviour under the form of a panic so we do need this.

Reply:
Upon further thought, I agree with your point here 👍 .